main.rb

```diff
@@ -260,50 +261,64 @@ def calculate_version_number(current_version, strategy, omit_zero,offset)
+  else
+    
+    if build_number_source.nil?
+      puts "Warning: No build number source specified. Skipping the build number update. If you want to update the build number, please set it in the Build Configuration.".yellow.bold
+    else
+      xcode_build_number = get_build_number(params, 'xcode')
+      puts "Project Build Number: #{xcode_build_number}"
+      puts "Appcircle Build Number: #{ac_env_build_number}" if build_number_source == 'env'
+      next_build_number = xcode_build_number
+
+      puts "Updating the Build number.".blue
+      current_build_number = get_build_number(params, build_number_source)
+      next_build_number = calculate_build_number(current_build_number, build_offset)
+      puts "Next build: #{next_build_number} Reason -> Source: #{build_number_source} offset: #{build_offset}"
+      increment_key(params, 'CFBundleVersion', next_build_number,'CURRENT_PROJECT_VERSION')
+      current_build_number = get_value_from_plist(params, 'CFBundleVersion','CURRENT_PROJECT_VERSION')
+    end 
+
+    if version_number_source.nil?
+      puts "Warning: No version number source specified. Skipping the version number update. If you want to update the version number, please set it in the Build Configuration.".yellow.bold
+    else
+      xcode_version_number = get_version_number(params, 'xcode')
+      puts "Project Version Number: #{xcode_version_number}"
+      next_version_number = xcode_version_number
+
+      puts "Updating the Version number.".blue
+      current_version_number = get_version_number(params, version_number_source)
+      if version_strategy == 'keep'
+        next_version_number = current_version_number
+      else
+        next_version_number = calculate_version_number(current_version_number, version_strategy, omit_zero, version_offset)
+      end
+      puts "Next version: #{next_version_number}  Reason -> Source: #{version_number_source} Strategy: #{version_strategy} Omit zero: #{omit_zero} Offset: #{version_offset}"
+      increment_key(params, 'CFBundleShortVersionString', next_version_number,'MARKETING_VERSION')
+      current_version_number = get_value_from_plist(params, 'CFBundleShortVersionString','MARKETING_VERSION')
+    end
+
+    puts "Build number updated to: #{current_build_number}".blue if build_number_source
+    puts "Version number updated to: #{current_version_number}".blue if version_number_source
 
-current_version_number = get_value_from_plist(params, 'CFBundleShortVersionString','MARKETING_VERSION')
-current_build_number = get_value_from_plist(params, 'CFBundleVersion','CURRENT_PROJECT_VERSION')
-puts "Build number updated to: #{current_build_number}"
-puts "Version number updated to: #{current_version_number}"
 
+    open(ENV['AC_ENV_FILE_PATH'], 'a') { |f|
+      f.puts "AC_IOS_NEW_BUILD_NUMBER=#{next_build_number}"
+      f.puts "AC_IOS_NEW_VERSION_NUMBER=#{next_version_number}"
```

Comment:
This code works incorrectly. 
If the user has selected a build number but not a version code, it will assign `nil` as `AC_IOS_NEW_VERSION_NUMBER`.